### ISSUE TITLE
deDE lockouts

### ISSUE THREAD

**Issue by muiqo** [Meivyn/AdventureGuideLockouts#2]
Thank you so much for picking up the AddOn! The default lockout-display is just awful... Some dungeons are incorrectly displayed as not completed in the deDE client, here are their InstanceInfos():

**Waycrest Manor**
[1]="Kronsteiganwesen",
[2]="party",
[3]=23,
[4]="Mythisch",
[5]=5,
[6]=0,
[7]=false,
[8]=1862,
[9]=5

**Shrine of the Storm**
[1]="Schrein des Sturms",
[2]="party",
[3]=23,
[4]="Mythisch",
[5]=5,
[6]=0,
[7]=false,
[8]=1864,
[9]=5

**Temple of Sethraliss**
[1]="Tempel von Sethraliss",
[2]="party",
[3]=23,
[4]="Mythisch",
[5]=5,
[6]=0,
[7]=false,
[8]=1877,
[9]=5

**Siege of Boralus**
[1]="Belagerung von Boralus",
[2]="party",
[3]=23,
[4]="Mythisch",
[5]=5,
[6]=0,
[7]=false,
[8]=1822,
[9]=5

**King's Rest**
[1]="Königsruh",
[2]="party",
[3]=23,
[4]="Mythisch",
[5]=5,
[6]=0,
[7]=false,
[8]=1762,
[9]=5

Haven't completed Freehold and Tol Dagor yet but here are their dumps anyway:

**Freehold**
[1]="Freihafen",
[2]="party",
[3]=23,
[4]="Mythisch",
[5]=5,
[6]=0,
[7]=false,
[8]=1754,
[9]=5

**Tol Dagor**
[1]="Tol Dagor",
[2]="party",
[3]=23,
[4]="Mythisch",
[5]=5,
[6]=0,
[7]=false,
[8]=1771,
[9]=5

**Comment by Meivyn** [comment]
Thanks for your report. I'll need more details on your issue, what's the problem with the lockouts? Some screenshots will be appreciated too. Your dumps will be used to fix it, but I need to understand what's working and what's not first.

Also be sure you do a ``/reload`` before telling me it doesn't work.

**Comment by muiqo** [comment]
![screenshot_1](https://user-images.githubusercontent.com/794518/44604215-7362bc00-a7e5-11e8-9ed9-ff2d47a43277.png)

![screenshot_2](https://user-images.githubusercontent.com/794518/44604288-aad16880-a7e5-11e8-815c-3fb427701fcf.png)

The listed dungeons(which are already completed) don't show any progress at all.

**Comment by Meivyn** [comment]
I see the issue. I'm gonna fix it in some minutes. If you see another instance that doesn't work, let me know.

**Comment by Meivyn** [comment]
Fixed in e0fb9bb5d7fe87e9a2a6e17e6b71f48c1a1cd295.

**Comment by muiqo** [comment]
Thanks for your work! :) After updating to the latest commit i'm getting this lua error

    1x ...nalSavedInstances\EncounterJournalSavedInstances-1.1.4.lua:209: Usage: EJ_GetEncounterInfo(ID)
    [C]: in function `EJ_GetEncounterInfo'
    ...nalSavedInstances\EncounterJournalSavedInstances-1.1.4.lua:209: in function <...nalSavedInstances\EncounterJournalSavedInstances.lua:107>
    ...nalSavedInstances\EncounterJournalSavedInstances-1.1.4.lua:521: in function <...nalSavedInstances\EncounterJournalSavedInstances.lua:520>
    [C]: in function `EncounterJournal_ListInstances'
    ...nalSavedInstances\EncounterJournalSavedInstances-1.1.4.lua:534: in function <...nalSavedInstances\EncounterJournalSavedInstances.lua:512>

    Locals:
    (*temporary) = nil

**Comment by Meivyn** [comment]
I don't get it. I don't have this error on my side, and line 209 is related to my World Bosses code. Did you killed any? You have only updated the addon and got this error? It wasn't showing before?

Try version [v1.1.3](https://github.com/Meivyn/EncounterJournalSavedInstances/releases/tag/v1.1.3) and [v1.1.4](https://github.com/Meivyn/EncounterJournalSavedInstances/releases/tag/v1.1.4) and let me know if the error is the same with both. 

This seems really weird to me, like it's not even in the same function of my last commit.

**Comment by muiqo** [comment]
It's super weird. When i'm reverting to Commit 3daff55(https://github.com/Meivyn/EncounterJournalSavedInstances/commit/3daff552cc809fa035a7fb6b4512fa18b1322b09) everything works fine without an error. Didn't kill any wordbosses this reset too :(

**Comment by exochron** [comment]
the german translation for "August Celestials" is missing.
```
if not wb.bosses[n].name then
	wb.bosses[n].name = EJ_GetEncounterInfo(wb.bosses[n].encounter)
end
```
Therefore it goes into the if and crashes, cause there is no encounter id for this in your worldBossesData.

**Comment by Meivyn** [comment]
This is what I thought at first but I don't get why he doesn't get this error in the precedent release.
 
But come to think of it, I noticed I was just creating a new table for other locales instead of replacing the values in current table. Thus ``L["August Celestials"]`` return ``nil`` in deDE locales and this wasn't intended.

I should just do:
```LUA
elseif locale == "deDE" then
    L["Shrine of the Storm"] = "Schrein des Sturms"
    L["Siege of Boralus"] = "Belagerung von Boralus"
    L["Temple of Sethraliss"] = "Tempel von Sethraliss"
    L["Waycrest Manor"] = "Kronsteiganwesen"
end
```
Instead of:
```LUA
elseif locale == "deDE" then
    L = {
        ["Shrine of the Storm"] = "Schrein des Sturms",
        ["Siege of Boralus"] = "Belagerung von Boralus",
        ["Temple of Sethraliss"] = "Tempel von Sethraliss",
        ["Waycrest Manor"] = "Kronsteiganwesen"
    }
end
```
I didn't notice before your answer. So thanks for your participation in this issue.

Will be fixed in next release.

**Comment by Meivyn** [comment]
Fixed in 779dfaa3c946af0b29b211e512d1a1fb42595b33.
